core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -299,6 +300,12 @@ class ControllerServer(
           setEligibleLeaderReplicasEnabled(config.elrEnabled)
       }
       controller = controllerBuilder.build()
+      // AutoMQ inject start
+      val controllerLeaderSupplier = new BooleanSupplier {
+        override def getAsBoolean: Boolean = controller.isActive
+      }
+      TelemetryLeaderRegistry.register(controllerLeaderSupplier)
```

Comment:
You could refer to `AutoMQApplication.registerSingleton(classOf[S3LogConfig], new KafkaS3LogConfig(config, null, kafkaRaftServer))` to setup MetricsConfig